nativelib/src/main/scala/scala/scalanative/native/time.scala

```diff
@@ -0,0 +1,27 @@
+package scala.scalanative
+package native
+
+@extern
+object time {
+
+  type time_t = CLong
+
+  @struct
+  class timespec(
+      val tv_sec: time_t = 0L, /* seconds */
+      val tv_nsec: CLong = 0L /* nanoseconds */
+  )
+
+  def clock_getres(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt  = extern
+  def clock_gettime(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt = extern
+  def clock_settime(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt = extern
```

Comment:
ok, given that there is not a way to achieve the desired resolution in a complete OS/independent way....
the question is: do you prefer to not support this kind of functions all around or you want to support OS/dependent compilation?

Unfortunately this apply to many other areas, files etc. etc. Posix systems are often not compatible with Windows.